modules/ipc/methods/base.js

```diff
@@ -243,5 +242,35 @@ module.exports = class BaseProcessor {
       err.message = err.message.replace('__method__', `"${payload.method}"`);
       payload.error = err;
     }
+
+    // only allow actions for authorized accounts
+    const methods = [
+      'eth_sendTransaction',
+      'eth_sign',
+      'eth_signTransaction',
+      'shh_post',
+      'shh_getMessages'
+    ];
+    if (methods.includes(payload.method)) {
+      const fromAddress = payload.params[0].from;
+      if (!this._isAdminConnection(conn)) {
+        const tab = db.getCollection('UI_tabs').findOne({ webviewId: conn.id });
+        if (
+          !tab ||
+          !tab.permissions ||
+          !tab.permissions.accounts ||
+          tab.permissions.accounts.length === 0 ||
+          !tab.permissions.accounts
+            .map(a => {
+              return a.toLowerCase();
+            })
+            .includes(fromAddress)
```

Comment:
this guy is a bit hairy - maybe abstract into a helper function, e.g. `if (!accountHasPermission(fromAddress) { ... }`, so it can get a couple unit tests of its own. wondering if this is the source of an unauthorized error I'm getting when I expect it to be kosher...